src/validate_pyproject/plugins/setuptools.schema.json

```diff
@@ -203,6 +203,8 @@
         "entry-points": {"$ref": "#/definitions/file-directive"},
         "optional-dependencies": {
           "type": "object",
+          "propertyNames": {"format": "pep508-identifier"},
```

Comment:
Hmm... Doesn't PEP 621 only say the _values_ need to be PEP 508 dependency specifiers? This seems to be related to the keys (property names).

All I can see about the keys in PEP 621 is 

> The keys MUST be valid values for the [Provides-Extra core metadata](https://packaging.python.org/specifications/core-metadata/#provides-extra-multiple-use).

and the documentation there says

> A string containing the name of an optional feature. Must be a valid Python identifier.

so if anything, this should be limited to valid Python identifiers..?

Reply:
Good catch 👍.

Yes that is right. (In terms of implementation pep508-identifier and python-identifier are the same thing, but I guess semantically the second is more precise).

Reply:
Pep 508 does also restrict possible values for the extra value though.

Reply:
Ah, right, okay. I was thinking `pep508-identifier` was `pep508-version-one`, but okay, `identifier` is just letters-or-digits-followed-by-spaces-and-dots-and-underscores-and-letters-and-digits. Looks good :)

Reply:
Sorry. Yesterday night I could not implement the fix. I was just changing the code :P.

I will add the changes directly in the upstream repo. Thank you very much @akx.